starfish/util/logging.py

```diff
@@ -0,0 +1,58 @@
+import os
+import platform
+from functools import lru_cache
+from json import JSONEncoder
+from subprocess import (
+    CalledProcessError,
+    check_output,
+)
+from typing import Mapping
+
+import pkg_resources
+
+import starfish
+from starfish.types import CORE_DEPENDENCIES
+
+
+@lru_cache(maxsize=1)
+def get_core_dependency_info() -> Mapping[str, str]:
+    dependency_info = dict()
+    for dependency in CORE_DEPENDENCIES:
+        version = get_dependency_version(dependency)
+        dependency_info[dependency] = version
+    return dependency_info
+
+
+def get_dependency_version(dependency: str) -> str:
+    return pkg_resources.get_distribution(dependency).version
+
+
+@lru_cache(maxsize=1)
+def get_git_commit_hash() -> str:
+    # First check if in starfish repo
+    os.chdir(os.path.dirname(starfish.__file__))
```

Comment:
You'll likely want a `finally` along with this `chdir` returning the user to `getcwd` otherwise file look ups will fail.

Reply:
+1